src/main/java/com/scalefocus/libraryproject/services/UserService.java

```diff
@@ -1,14 +1,21 @@
 package com.scalefocus.libraryproject.services;
 
 import com.scalefocus.libraryproject.entities.UserEntity;
+import com.scalefocus.libraryproject.exceptions.RegisterException;
 import com.scalefocus.libraryproject.models.UserModel;
 import com.scalefocus.libraryproject.repositories.UserRepository;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
 import org.springframework.stereotype.Service;
 
+import java.time.ZonedDateTime;
+
+import static com.scalefocus.libraryproject.enums.Role.USER;
+
 @Service
 public class UserService {
     private UserRepository userRepository;
+    private BCryptPasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
```

Comment:
In main, we added a BCryptPasswordEncoder spring bean. Please, resolve the conflict and use it instead.